src/core/withExperiment.js

```diff
@@ -8,15 +8,54 @@ import log from 'core/logger';
 import tracking from 'core/tracking';
 import { getDisplayName } from 'core/utils';
 
+/*  Usage
+ *
+ *  To add experiment support to a component, do the following:
+ *
+ *  1. Include `withExperiment()` in the `compose` function of the component.
+ *  2. Set a unique `id` for the experiment in the call to `withExperiment`
+ *     (see example below).
+ *  3. Define a set of variants for the experiment in the call to
+ *     `withExperiment`, each of which will be assigned an `id` and a
+ *     `percentage` which represents the portion of users who will be assigned
+ *     this variant (see example below).
+ *
+ *  Note: The sum of all `percentage` values must be exactly 1. An exception
+ *        will be thrown if that is not the case.
+ *
+ *  Note: To create a portion of the user population who will not be enrolled
+ *        in the experiment, assign a percentage to the special
+ *        `NOT_IN_EXPERIMENT` variant.
+ *
+ *  Example:
+ *
+ *     withExperiment({
+ *      id: 'some-experiment-id',
+ *      variants: [
+ *        { id: 'variant-a', percentage: 0.2 },
+ *        { id: 'variant-b', percentage: 0.2 },
+ *        { id: NOT_IN_EXPERIMENT, percentage: 0.6 },
```

Comment:
I am wondering why this needs to be a special variant and therefore why we have a `isUserInExperiment` prop created in the HOC, which might not be useful in all experiments.

I am wondering why we would need all of this in a generic-ish HOC instead of declaring a "not_in_experiment" variant for a given experiment and use `const isUserInExperiment = variant !== "not_in_experiment";`, just like we would have "variant_blue" and `const isBlue = variant === "variant_blue";'`, etc.

Reply:
Because `withExperiment` controls sending an enrollment event, and we don't send an enrollment event for users who are not in the experiment, so `withExperiment` needs to know if a user has been put into the special branch that represents users not in the experiment.

Reply:
Ah, I see. That makes sense then.